core/src/test/java/dk/alexandra/fresco/framework/util/TestDetermSecureRandom.java

```diff
@@ -38,18 +41,54 @@ public void testSetSeed() throws NoSuchAlgorithmException {
   
   @Test(expected = NoSuchAlgorithmException.class)
   public void testNonExistingAlgorithm() throws NoSuchAlgorithmException {
-    new DetermSecureRandom(new byte[]{0x01}, "Bla");        
+    new HMacDRBG(new byte[]{0x01}, "Bla");        
+  }
+  
+  @Test(expected = MPCException.class)
+  public void testInvalidKey() throws NoSuchAlgorithmException {
+    HMacDRBG m = new HMacDRBGFakeKeySpec();
+    byte[] randBytes1 = new byte[10];
+    m.nextBytes(randBytes1);
```

Comment:
Change to:
HMacDRBG m = new HMacDRBG();
m.getSafeKey(()->...broken key supplier...)
No need for inheritance

Reply:
There is actually. The method call which throws the InvalidKey exception is mac.init(). This I don't want to expose as that is very internal. Therefore the current solution.

Reply:
In that case your design for testability should be reworked.
Create a method called safeInit(Consumer<SecretKey) which does the exception handling and nothing else. Supply if with max::init

Reply:
I don't understand the supply if with max::init comment. But I'll look into it.

Reply:
Please check the updated comment: Actually it could just be a runnable or another simple interface.
 
Hence create a method safeInit(Callable) that calls the callable and guarded for the exception

Reply:
Seems like the Consumer interface cannot handle methods which casts checked exceptions (as this one does), so that does not seem to be the solution. Correct me if I'm wrong. Havn't really worked with the functional style before.

Reply:
As you might have realized, your latest comment is not in sync